docs/resources/jwt_issuer.md

```diff
@@ -0,0 +1,64 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "cockroach_jwt_issuer Resource - terraform-provider-cockroach"
+subcategory: ""
+description: |-
+  Configuration to manage external JWT Issuers to access CockroachDB Cloud API via JWT.
+---
+
+# cockroach_jwt_issuer (Resource)
+
+Configuration to manage external JWT Issuers to access CockroachDB Cloud API via JWT.
+
+## Example Usage
+
+```terraform
+resource "cockroach_jwt_issuer" "example" {
+  issuer_url = "https://accounts.google.com"
+  audience   = "test_audience"
+  jwks       = "{\"keys\":[{\"alg\":\"RS256\",\"e\":\"AQAB\",\"kid\":\"test_kid1\",\"kty\":\"RSA\",\"n\":\"09lq1lCEuteonwDJOhGTDak11ThplZuC9JEWQNdBnBSQwlkJQIE7A7nTBO0xTibcsh2HwYkC-N_Gs1jP4iwN3dRqnu5FwG2ct5mY8KLwJiHzToFC0MKenSFQCy0FviNtOnpiObcUlDvR2NDeNtMl_6SPzcQEt7GUTBBYZgoAxPmOgevki6ZNO6Y86xFqx3y6v8EPwW010AiC60r4AHGCTBhYF4uqmq5JH2UU4dDh9Udc-9LZxlSqPwJvnKDG2GjcnD8TsU3wjfEM_nRmx3dnXsrZUXYfNGtdv5dlHywf5AhkJmTavqcsJkgrNA-PNBghFMcCR816_kCIkCYWLWC5vQ\"}]}"
+  claim      = "email"
+  identity_map = [
+    {
+      token_identity = "token_identity"
+      cc_identity    = "cc_identity"
+      is_regex       = false
+    },
+    {
+      token_identity = "(.*)"
+      cc_identity    = "\\1@example.com"
+      is_regex       = true
+    },
+  ]
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `audience` (String) The intended audience for consuming the JWT.
+- `issuer_url` (String) The URL of the server issuing JWTs.
+
+### Optional
+
+- `claim` (String) Used to identify the user from the external Identity Provider. Defaults to "sub".
+- `identity_map` (Attributes List) A list of mappings to map the external token identity into CockroachDB Cloud. (see [below for nested schema](#nestedatt--identity_map))
+- `jwks` (String) A set of public keys (JWKS) used to verify the JWT.
+
+### Read-Only
+
+- `id` (String) The unique identifier of the JWT Issuer resource.
+
+<a id="nestedatt--identity_map"></a>
+### Nested Schema for `identity_map`
+
+Required:
+
+- `cc_identity` (String) Specifies how to map the fetched token identity into CockroachDB Cloud.
+- `token_identity` (String) Specifies how to fetch external identity from the token claim.
+
+Optional:
+
+- `is_regex` (Boolean) Specifies whether the token_identity and cc_identity configurations are in regex format.
```

Comment:
I realized that the is_regex is baked into the api so if we wanted to change that we'd probably want to do it in the api as well.   It seems like a good Idea to me but it may not be worth the effort.  What do you all think?

Reply:
I agree that `is_regex` is redundant, but we could only do away with `token_identity` and `cc_identity`.

Reply:
I'm not sure I follow.  It's not currently redundant right?  It's necessary because we don't know if an identity should be interpreted as a string or a regex.  I'm suggesting instead of verifying it that way that we instead of different field names based on whether its a string or a regex.  Note that I've mixed the usages of these here so illustrate the additional power that it provides.  For example:

```
  identity_map = [
    {
      token_identity = "token_identity"
      cc_identity_pattern    = "\\1@example.com"
    },
    {
      token_identity_pattern = "(.*)"
      cc_identity    = "cc_identity"
    },
```

Reply:
I see, got it.
However, it is possible to derive whether the input has a regex.
As it turns out, the identity map implementation within CRDB also works this way (without needing to specify whether it's a regex): https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/pgwire/identmap/ident_map.go

Reply:
oh I see, that is checking to see if the first token is a slash.  

> If the system-identity field starts with a slash, it will be interpreted as a regular expression

That seems like a better approach as long as a leading slash isn't a valid value for an identity.

Reply:
Created a PR in managed-service for the same: https://github.com/cockroachlabs/managed-service/pull/18531